### ISSUE TITLE
Sending Article version and correction data to Crossref

### ISSUE THREAD

**Issue by Melissa37** [elifesciences/elife-crossref-feed#145]
## Problem / Motivation
Production wants to send information about corrections and retractions to Crossref so this information is available to users of their API and we are helping with the need to track correction/retraction metadata

## Proposed solution
eLife add CrossMark metadata to our deposits: 

**Corrections/Retractions**
Crossref documentation: https://www.crossref.org/education/crossmark/crossmark-registering-updates/
The metadata for the update is part of the Crossmark metadata, and should include a link to the item being updated, and the type of update:

```
<updates>
<update type="retraction" label="Retraction" date="2009-09-14">10.5555/12345678</update>
</updates>
```

```
<updates>
<update type="correction" label="Correction" date="2012-02-29">10.5555/12345679</update>
</updates>
```

My assumption from their example XML files is that when we publish a retraction or correction we add this metadata to the retraction notice and leave the original article as is. The DOI in this update is the DOI of the original article, so it will pick it up.

**New versions**
From Crossref:
>Example 3: in-situ correction
When a member does not issue a separate update/correction/retraction notice and instead just makes the change to the document (without changing its DOI either), this is called an in-situ update. In-situ updates or corrections are not recommended because they tend to obscure the scholarly record. How do you tell what the differences are between what you downloaded and the update? How do you differentiate them when citing them (remember, we are only talking about “significant updates” here)? However, some members need to support in-situ updates, and this is how they can be supported.

```
<updates>
<update type="correction" label="Correction" date="2012-05-12">10.5555/12345681</update>
</updates>
```

This is added to the main XML of the article. 

**From Crossref re the need for a DOI** 
>As we advise for users supplying Crossmark data simply as a workaround to make their free-to-read content visible in our API, you could just use the DOI that's being updated in the <crossmark_policy> tags as well as in the <doi_data>. 
The later is more of a 'hack' than the former, but since we're in the midst of changing approaches to this kind of update metadata, it's still a fine option.  If you want to register a DOI for your updates policies that's great, but we don't want that extra step to discourage anyone from sending in the updates metadata. 

```
<crossmark>
<crossmark_policy>10.5555/crossmark_policy</crossmark_policy>
<crossmark_domains>
<crossmark_domain>
<domain>psychoceramics.labs.crossref.org</domain>
</crossmark_domain>
</crossmark_domains>
<crossmark_domain_exclusive>true</crossmark_domain_exclusive>
```

## Clarification needed and assumptions
- assume we will need to register our whole archive with CrossMark in order to ensure all corrections/retractions will link correctly, and update Correction and retraction metadata as well.

>Deposit Crossmark metadata - as part of your regular Crossref metadata deposit, and can also be deposited as stand-alone data to populate backfiles. For Crossmark-only deposits, see the schema and schema documentation relating to resource-only deposits.
https://www.crossref.org/education/metadata-stewardship/maintaining-your-metadata/adding-metadata-to-an-existing-record/

This does not seem to support our use case and we'd be better off just re-depositing everything?

Question: '<crossmark_domain_exclusive>true</crossmark_domain_exclusive>'. What does this refer to? Is eLife True or False?

---

## Tasks

- [x] Add Crossmark basics to deposit XML, including DTD and minimal tagging
- [x] Test minimal Crossmark deposit XML against schema for validation
- [x] What tags are mandatory? crossmark_policy  **Answer:** Yes, crossmark_policy is mandatory, but the article's DOI can be used for its value if another policy page is not available to be used
- [x] What tags are mandatory? crossmark_domain / crossmark_domains **Answer:** They are optional
- [x] What tags are mandatory? crossmark_domain_exclusive **Answer:** They are optional
- [x] Crossref generation library configuration file preference for adding Crossmark output (PR https://github.com/elifesciences/elife-bot-formula/pull/60)
- [x] Correction and Retraction articles, Crossmark XML deposit output for these
- [x] Test Correction / Retraction article deposit for schema validation
- [x] Deposit one eLife Correction deposit with Crossmark included and check the result - may also require redepositing the article it is correction / retracting
- [x] If happy, deploy code for depositing Correction / Retraction in deposits
- [x] ~~Potentially redeposit all eLife articles to Crossref so they are all enrolled in the Crossmark program (if this is how it works)~~ No, I don't think we need to redeposit all articles, just the pertinent ones
- [x] Re-deposit correction and retraction articles - Completed June 9, 2020
- [ ] Add in-situ deposit logic (if it is clear what to do)
- [ ] Expand todo task list to include: steps to add each version of an article as Crossmark data

## Technical notes
- We could use this technology
- More notes or suggestions

**Comment by gnott** [comment]
Here are some of my notes and thoughts, for discussion:


- I think structured abstracts could possibly be added to the article data structure used by the Crossref generation library without involving integration with other data schemas

- Clinical trial data would be added as a new property of an `Article` object, and then we can include that in Crossref deposits

Crossmark related:

- Code in the old, archived, Crossref generation library: For defining the Crossmark policy and domain (https://github.com/elifesciences/elife-poa-xml-generation/blob/develop/generateCrossrefXml.py#L34-L35)
- Old code that added Crossmark XML to a Crossref deposit, but it was never used for real I think https://github.com/elifesciences/elife-poa-xml-generation/blob/develop/generateCrossrefXml.py#L34-L35
- Perhaps not all articles would need to be deposited with Crossmark data, but my guess is if we want to register a Correction, for example, the article that is being corrected would need to be deposited with Crossmark, and then the correction article as well afterward

XML and testing

- For clinial trials support, need to add XML schema prefix to the Crossref XML deposit, e.g. `xmlns:ct="http://www.crossref.org/clinicaltrials.xsd"`
- Add additional settings for Crossmark into the `elifecrossref` library `.cfg` file to turn on/off Crossmark deposits, specify the Crossmark domain and Crossmark policy DOI
- For Crossmark, test examples for XML validity, to check all mandatory tags are included in the code library

**Comment by gnott** [comment]
I will now update the first comment with some todo task checkboxes, since nothing is there yet.

I wonder @Melissa37 if the clinical trials data might be better split off into a new issue? I'm not sure I see how it connects with Crossmark, and it is also blocked by the structured abstract changes.

**Comment by Melissa37** [comment]
Sure, that's a good idea. I will split them out.

**Comment by Melissa37** [comment]
Linked to https://github.com/elifesciences/issues/issues/5641

**Comment by gnott** [comment]
To note, in testing to validate against the Crossref schema, if Crossmark data is included, the `< crossmark_policy>` tag is required. 

@Melissa37 would eLife create a separate web page and/or DOI for this, or is there an existing page to be used?

Years ago in the Crossmark code that was not yet used, I had `10.7554/eLife/crossmark_policy` for this value (which would end up as `https://doi.org10.7554/eLife/crossmark_policy`).

Page at Crossref regarding the policy page https://www.crossref.org/education/crossmark/crossmark-policy-page/

**Comment by Melissa37** [comment]
@gnott 

Sorry if it was not clear in the top thread:

>As we advise for users supplying Crossmark data simply as a workaround to make their free-to-read content visible in our API, you could just use the DOI that's being updated in the <crossmark_policy> tags as well as in the <doi_data>.
The later is more of a 'hack' than the former, but since we're in the midst of changing approaches to this kind of update metadata, it's still a fine option. If you want to register a DOI for your updates policies that's great, but we don't want that extra step to discourage anyone from sending in the updates metadata.

Does that make sense? We don't need to create this new page with a DOI, but use the DOI of the article in hand as the <crossmark_policy>10.5555/crossmark_policy</crossmark_policy>

**Comment by gnott** [comment]
I failed to read it all sufficiently to remember that point, thanks @Melissa37!

For the Crossref generator logic, I will still want to make it possible to specify a single value for `<crossmark_policy>` but if that is not specified, it can fallback to using the article's DOI.

**Comment by gnott** [comment]
The older code I started from also included the `<crossmark_version>` tag. This tag is still present in the Crossref schema, but I cannot find much current documentation for what values to specify in it, or its significance. I think it may have to do with versioning of content and for displaying buttons, which is not our primary goal here -- it being to send corrections and retractions first -- so I'm going to omit the `<crossmark_version>` tag for now until how it should be used is clearer.

**Comment by gnott** [comment]
Looking for a little more clarity about what the Crossmark domain tag values are about, it looks like the documention in https://www.crossref.org/schemas/common4.4.1.xsd is the most explanation I can find. I will include it here for reference:

**`<crossmark_domain_exclusive>`**

"Some publishers encourage broad third party hosting of the publisher's content. Other publishers do not. And still others vary their policy depending on whether a particular article has been published under an OA policy or not. This boolean flag allows the publisher to indicate whether the CrossMarked content will only legitimately be updated on the CrossMark domain (true) or whether the publisher encourages updating the content on other sites as well (false)."

**`<crossmark_domains>`**

"Container element for crossmark_domain. A list of domains where the publisher maintains updates and corrections to their content. Minimally, one of these should include the Internet domain name of the publisher's web site(s), but the publisher might also decide to include 3rd party aggregators (e.g. Ebsco, IngentaConnect) or archives with which the publisher has agreements to update the content"

**`<crossmark_domain>`**

"This should be a simple Internet domain name or subdomain name (e.g. www.psychoceramics.org or psychoceramics.org). It is used to identify when a referring URL is coming from a CrossMark domain. A "crossmark_domain" is made up of two subelements; a "domain" and a "filter". The domain is required but the filter is optional and is only needed for use in situations where content from multiple publishers/publications is on the same host with the same domain name (e.g. an aggregator) and one needs to use the referrer's URI "path" to further determine whether the content in a crossmark domain."

**`<domain>`**

"Required element. This should be a simple Internet domain name or subdomain name (e.g. www.psychoceramics.org or psychoceramics.org). It is used to identify when a referring URL is coming from a CrossMark domain."

**`<filter>`**

"Optional element. The filter element is used to disambiguate content in situations where multiple publishers share the same host (e.g. when on an aggregated platform). It should contain a substring of the path that can be used to uniquely identify a publisher's or publication's content. For instance, using the string "alpsp" here would help the CrossMark system distinguish between ALPSP publications on the ingentaconnect host and other publications on the same host."

**Comment by gnott** [comment]
For eLife's situation, I think the following will be suitable, and is what we can generate for now:

```
<crossmark_domains>
  <crossmark_domain>
    <domain>elifesciences.org</domain>
  </crossmark_domain>
</crossmark_domains>
<crossmark_domain_exclusive>true</crossmark_domain_exclusive>
```

My interpretation is: any Crossref article on `elifesciences.org` can be used to identify referring URLs in the Crossmark program (or something like that), and no "filter" is required because it is not a shared journal hosting site. The `"true"` for the `crossmark_domain_exclusive` value seems easiest, that the legitimate updates to articles would happen on `elifesciences.org`, and if any mirror sites for eLife content exists, they may not receive updated content.

**Comment by gnott** [comment]
I'm not entirely sure about the domain exclusive value though, @Melissa37 what is your thought here for eLife's values? When I read above "... whether the publisher encourages updating the content on other sites as well (false)", I would assume eLife would encourage other sites to update their content, so maybe eLife should use `false` as the value in deposits?

**Comment by Melissa37** [comment]
>The older code I started from also included the <crossmark_version> tag. This tag is still present in the Crossref schema, but I cannot find much current documentation for what values to specify in it, or its significance. I think it may have to do with versioning of content and for displaying buttons, which is not our primary goal here -- it being to send corrections and retractions first -- so I'm going to omit the <crossmark_version> tag for now until how it should be used is clearer.

Sounds good to me!

Thanks for the further digging!

><crossmark_domain_exclusive>
"Some publishers encourage broad third party hosting of the publisher's content. Other publishers do not. And still others vary their policy depending on whether a particular article has been published under an OA policy or not. This boolean flag allows the publisher to indicate whether the CrossMarked content will only legitimately be updated on the CrossMark domain (true) or whether the publisher encourages updating the content on other sites as well (false)."

So, we are False, even though others rarely update we know PMC does


><crossmark_domains>
"Container element for crossmark_domain. A list of domains where the publisher maintains updates and corrections to their content. Minimally, one of these should include the Internet domain name of the publisher's web site(s), but the publisher might also decide to include 3rd party aggregators (e.g. Ebsco, IngentaConnect) or archives with which the publisher has agreements to update the content"

So, based on what we know, PMC and PubMed are the only domains that accept our versioned content, right?

><crossmark_domain>
"This should be a simple Internet domain name or subdomain name (e.g. www.psychoceramics.org or psychoceramics.org). It is used to identify when a referring URL is coming from a CrossMark domain. A "crossmark_domain" is made up of two subelements; a "domain" and a "filter". The domain is required but the filter is optional and is only needed for use in situations where content from multiple publishers/publications is on the same host with the same domain name (e.g. an aggregator) and one needs to use the referrer's URI "path" to further determine whether the content in a crossmark domain."

I assume we don't need the filter, just the domain?

Thanks @gnott

**Comment by gnott** [comment]
I continued to add `<updates>` and `<update>` tag to the logic for a correction article as a first attempt, and I have a deposit which is valid against the Crossref schema validation tool.

The `label="Correction"` attribute is not allowed in more recent schemas I guess, since on their documentation page the samples use Crossref schema version `4.3.0`, my sample using version `4.4.1` the `@label` for the `<update>` tag was refused as not allowed.

**Comment by Melissa37** [comment]
@gnott Shall I contact Crossref support about this issue:

>The label="Correction" attribute is not allowed in more recent schemas I guess, since on their documentation page the samples use Crossref schema version 4.3.0, my sample using version 4.4.1 the @label for the <update> tag was refused as not allowed.

It could be a mistake or intentional and the intention is not clear to us! 

Please let me know?

Thanks!

M

**Comment by gnott** [comment]
I looked at the schema files in more detail, in version `4.3.0` (https://www.crossref.org/schemas/common4.3.0.xsd) the `label ` (lowercase) attribute is listed and required:

`<xsd:attribute name="label" use="required" type="cm_update_label">`

It's not present in version `4.4.1` (https://www.crossref.org/schemas/common4.4.1.xsd), and the `cm_update_label` is mentioned in the change history, so it looks like it was removed:

`4.3.5	(PDF) 10/9/2014 ..., restricted CrossMark update types to an enumerated list, removed cm_update_label`

**Comment by gnott** [comment]
Discussion today: No need to involve Crossref about the label attribute that was removed from their schema, it is just old documentation of theirs.

The test deposits of correction and retraction articles are a-ok, so the next step will be to deploy the code to enable Crossmark deposits, then re-deposit all the previous correction and retraction articles.

Next after that is to discuss _in situ_ corrections more, and depositing clinical trials (which will be actionable next I'm not sure).

**Comment by gnott** [comment]
Seeing as the remaining todo tasks are covered in a new issue https://github.com/elifesciences/elife-crossref-feed/issues/147, @Melissa37 do you think this older issue may be something we would be able to close, or is there more here that was not addressed?

**Comment by Melissa37** [comment]
No that's cool lets close it, thanks!
M

**Comment by gnott** [comment]
Thanks @Melissa37! For the purposes of this issue, we added Crossmark support, it includes the ability to send formal correction and retraction article data, and also clinical trials data. The inclusion of in-situ updates into the Crossmark data is not yet added, and can be continued on the separate open issue https://github.com/elifesciences/elife-crossref-feed/issues/147.
